daemons/based/based_callbacks.c

```diff
@@ -1440,6 +1440,23 @@ cib_process_command(xmlNode *request, const cib__operation_t *operation,
             cib_read_config(config_hash, result_cib);
         }
 
+        /* @COMPAT Nodes older than feature set 3.19.0 don't support
+         * transactions. In a mixed-version cluster with nodes <3.19.0, we must
+         * sync the updated CIB, so that the older nodes receive the changes.
+         * Any node that has already applied the transaction will ignore the
+         * synced CIB.
+         *
+         * To ensure the updated CIB is synced from only one node, we sync it
+         * from the originator.
+         */
+        if ((operation->type == cib__op_commit_transact)
+            && pcmk__str_eq(originator, OUR_NODENAME, pcmk__str_casei)
+            && compare_version(crm_element_value(the_cib, XML_ATTR_CRM_VERSION),
```

Comment:
@kgaillot Oh yeah I meant to ask -- checking the feature set in the CIB itself is enough, right?

I ask because we set enable legacy mode if we receive a non-replace op with `F_CIB_GLOBAL_UPDATE` set... instead of just checking the feature set in the CIB, for some reason.

If I'm missing something, then we could set a flag upon receiving any request with a feature set less than 3.19.0

Reply:
The only possible complication I can think of is the first CIB sync with the lower feature set at startup. That shouldn't matter here.